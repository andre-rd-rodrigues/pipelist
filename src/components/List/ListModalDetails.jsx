import React from "react";
import { Col, Row } from "react-bootstrap";

const ListModalDetails = ({ label, value }) => {
  return (
    <Row className="align-items-baseline justify-content-center">
      <Col sm={4} md={4} lg={4}>
        <div className="person_modal_details_field_label">
          <h5>{label}</h5>
        </div>
      </Col>
      <Col sm={8} md={8} lg={8}>
        <p>{value}</p>
      </Col>
    </Row>
  );
};

export default ListModalDetails;
