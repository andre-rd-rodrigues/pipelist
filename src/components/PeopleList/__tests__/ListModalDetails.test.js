import { render, screen } from "@testing-library/react";
import reactTestRenderer from "react-test-renderer";
import ListModalDetails from "../ListModalDetails";

describe("List modal details", () => {
  it("should render correctly", () => {
    const wrapper = reactTestRenderer.create(<ListModalDetails />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render according to props", () => {
    render(<ListModalDetails label="Assistant" value="Stacy Conely" />);
    expect(screen.getByText(/Stacy Conely/i)).toBeInTheDocument();
  });
});
