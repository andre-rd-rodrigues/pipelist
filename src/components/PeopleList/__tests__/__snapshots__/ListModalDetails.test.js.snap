// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List modal details should render correctly 1`] = `
<div
  className="align-items-baseline justify-content-center row"
>
  <div
    className="col-lg-4 col-md-4 col-sm-4"
  >
    <div
      className="person_modal_details_field_label"
    >
      <h5 />
    </div>
  </div>
  <div
    className="col-lg-8 col-md-8 col-sm-8"
  >
    <p />
  </div>
</div>
`;
